File: src/dataset/Java/armstrong_numbers_Exercism_java_ValMati/claude/ClaudeSonnet4_ArmstrongNumbers_v3.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {
        if (numberToCheck < 0) return false;
        if (numberToCheck < 10) return true;
        
        int original = numberToCheck;
        int digitCount = 0;
        int temp = numberToCheck;
        
        while (temp > 0) {
            digitCount++;
            temp /= 10;
        }
        
        int calculatedNumber = 0;
        temp = numberToCheck;
        
        while (temp > 0) {
            int digit = temp % 10;
            calculatedNumber += pow(digit, digitCount);
            temp /= 10;
        }
        
        return original == calculatedNumber;
    }
    
    private int pow(int base, int exponent) {
        int result = 1;
        for (int i = 0; i < exponent; i++) {
            result *= base;
        }
        return result;
    }
}

File: src/dataset/c/armstrong-numbers_exercism-Manoloon/claude/ClaudeSonnet4_armstrong_numbers_v3.c
#include "armstrong_numbers.h"
#include <stdint.h>

bool is_armstrong_number(int candidate)
{
    if(candidate < 0) return false;
    if(candidate < 10) return true;
    
    int original = candidate;
    int digits = 0;
    int temp = candidate;
    
    while(temp > 0) {
        digits++;
        temp /= 10;
    }
    
    int result = 0;
    temp = candidate;
    
    while(temp > 0) {
        int digit = temp % 10;
        int power = 1;
        for(int i = 0; i < digits; i++) {
            power *= digit;
        }
        result += power;
        temp /= 10;
        
        if(result > original) return false;
    }
    
    return result == original;
}